mango-core/src/main/java/org/calrissian/mango/types/TypeRegistry.java

```diff
@@ -117,7 +117,7 @@ public Object decode(String alias, U value) {
         if (encoder != null)
             return encoder.decode(value);
 
-        throw new TypeDecodingException("An unknown alias [" + value + "] was encountered");
+        throw new TypeDecodingException("An unknown alias [" + alias + "] was encountered while decoding value [" + value + "]. Valid aliases are " + String.join(", ", aliasMapping.keySet()));
```

Comment:
Yes, but printing the value doesn't add anything to the value of the error message when the alias is the reason for the exception so lets not do that.  If the exception was from the act of decoding then yes it would potentionally make sense to try to show something about the value.

Oops, I meant Java 7.  Guava Joiner is fine.